crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -7206,10 +7214,23 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
 
         let operand_type = self.infer_expression(operand);
 
+        self.infer_unary_expression_impl(*op, operand_type, unary)
+    }
+
+    fn infer_unary_expression_impl(
```

Comment:
nit: I think our pattern elsewhere has generally been to use the `_type` prefix for methods like this rather than `_impl`. E.g. `infer_binary_expression` calls `infer_binary_expression_type` which is an inner method that recursively calls itself; `infer_compare_expression` calls `infer_binary_type_comparison` (not a prefix, but a similar-ish principle!)